pkg/workload/admissionchecks.go

```diff
@@ -93,12 +98,30 @@ func ResetChecksOnEviction(w *kueue.Workload) {
 		checks[i] = kueue.AdmissionCheckState{
 			Name:               checks[i].Name,
 			State:              kueue.CheckStatePending,
-			LastTransitionTime: metav1.NewTime(time.Now()),
+			LastTransitionTime: metav1.NewTime(realClock.Now()),
 			Message:            "Reset to Pending after eviction. Previously: " + string(checks[i].State),
 		}
 	}
 }
 
+func ResetChecksOnDeactivation(w *kueue.Workload) bool {
```

Comment:
I don't think we need a new function for this, it is virtually the same as `ResetChecksOnEviction`, the only differences being `time` as a new arg and skipping of updating for Pending. 

So, I would suggest to just update the existing function and reuse in this context. Deactivation is a special case of Eviction, so I think the name is also fine.